Design.md

```diff
@@ -253,3 +262,23 @@ Get the platform usage of global (without query parameter), tenant (with tenant_
 ## S3 Operations Example Activity Diagram
 ![S3 Object operations](Design_Files/OSE-OSIS-Object-operations.png)
 
+## Cache Design
+
+1. Implement Cache with following principles:
+    1. Cache with a maximum capacity.
+    1. Cache should have LRU (Least Recently Used) eviction policy.
+    1. Each cache entry must have an expiration time (TTL). 
+        1. Cache `put` call should include `expirationTime` for the entry.
+        1. Use `ScheduledFuture` in java for cache entry removal after TTL.
+    1. Cache must handle concurrency.
+        1. Use `Read locks` for `get` calls and `write locks` for `put` calls
```

Comment:
Can you explain why we need to use locks(mutex) for put and get calls? Are we talking about S3 and IAM calls?

Reply:
As this cache is using one common resource across all JVM threads, to maintain a synchronized behavior we are using locks